persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_51_gsheet/generated/entities/persist_client.bal

```diff
@@ -0,0 +1,497 @@
+            outer join var workspace in workspacesStream on ['object.workspaceWorkspaceId] equals [workspace?.workspaceId]
+            do {
+                return {
+                    ...'object,
+                    "department": department,
+                    "workspace": workspace
+                };
+            };
+        if unionResult is error {
+            return <persist:Error>error(unionResult.message());
+        }
+        return <persist:InvalidKeyError>error("Invalid key: " + key.toString());
+    }
+
+    private isolated function queryEmployeesStream(EmployeeTargetType targetType = <>) returns stream<targetType, error?> = @java:Method {
+        'class: "io.ballerina.stdlib.persist.datastore.GoogleSheetsProcessor",
+        name: "queryStream"
+    } external;
+
+    isolated resource function get workspaces(WorkspaceTargetType targetType = <>) returns stream<targetType, persist:Error?> = @java:Method {
+        'class: "io.ballerina.stdlib.persist.datastore.GoogleSheetsProcessor",
+        name: "query"
+    } external;
+
+    isolated resource function get workspaces/[string workspaceId](WorkspaceTargetType targetType = <>) returns targetType|persist:Error = @java:Method {
+        'class: "io.ballerina.stdlib.persist.datastore.GoogleSheetsProcessor",
+        name: "queryOne"
+    } external;
+
+    isolated resource function post workspaces(WorkspaceInsert[] data) returns string[]|persist:Error {
+        _ = check self.persistClients.get(WORKSPACE).runBatchInsertQuery(data);
+        return from WorkspaceInsert inserted in data
+            select inserted.workspaceId;
+    }
+
+    isolated resource function put workspaces/[string workspaceId](WorkspaceUpdate value) returns Workspace|persist:Error {
+        _ = check self.persistClients.get(WORKSPACE).runUpdateQuery(workspaceId, value);
+        return self->/workspaces/[workspaceId].get();
+    }
+
+    isolated resource function delete workspaces/[string workspaceId]() returns Workspace|persist:Error {
+        Workspace result = check self->/workspaces/[workspaceId].get();
+        _ = check self.persistClients.get(WORKSPACE).runDeleteQuery(workspaceId);
+        return result;
+    }
+
+    private function queryWorkspaces(string[] fields) returns stream<record {}, persist:Error?>|error {
+        stream<Workspace, error?> workspacesStream = self.queryWorkspacesStream();
+        stream<Building, error?> buildingsStream = self.queryBuildingsStream();
+        record {}[] outputArray = check from record {} 'object in workspacesStream
+            outer join var building in buildingsStream on ['object.locationBuildingCode] equals [building?.buildingCode]
+            select persist:filterRecord({
+                ...'object,
+                "building": building
+            }, fields);
+        return outputArray.toStream();
+    }
+
+    private function queryOneWorkspaces(anydata key) returns record {}|error {
```

Comment:
error type was changed to InvalidKeyError

Reply:
https://github.com/ballerina-platform/module-ballerina-persist/blob/e459c7b8cc8656c229e52bb233c0d6f97fbb8d0e/ballerina/tests/googlesheets_rainier_generated_client.bal#L348